integration_test/gce/gce_testing.go

```diff
@@ -451,29 +465,45 @@ func firstTrace(it *trace.TraceIterator) (*cloudtrace.Trace, error) {
 	return trace, nil
 }
 
-// WaitForMetric looks for the given metric in the backend and returns it if it
+// WaitForMetric uses WaitForMetricSeries to return a single metric or appropriate error
+func WaitForMetric(ctx context.Context, logger *log.Logger, vm *VM, metric string, window time.Duration, extraFilters []string, isPrometheus bool) (*monitoringpb.TimeSeries, error) {
+	series, err := WaitForMetricSeries(ctx, logger, vm, metric, window, extraFilters, isPrometheus, 1)
+	if err != nil {
+		return nil, err
+	}
+	logger.Printf("WaitForMetric metric=%v, series=%v", metric, series)
+	return series[0], nil
+}
+
+// WaitForMetricSeries looks for the given metrics in the backend and returns it if it
 // exists. An error is returned otherwise. This function will retry "no data"
 // errors a fixed number of times. This is useful because it takes time for
 // monitoring data to become visible after it has been uploaded.
-func WaitForMetric(ctx context.Context, logger *log.Logger, vm *VM, metric string, window time.Duration, extraFilters []string, isPrometheus bool) (*monitoringpb.TimeSeries, error) {
+// A return value of (nil, nil) indicates that the evaluation succeeded
+// but returned no data.
+func WaitForMetricSeries(ctx context.Context, logger *log.Logger, vm *VM, metric string, window time.Duration, extraFilters []string, isPrometheus bool, minimumRequiredSeries int) ([]*monitoringpb.TimeSeries, error) {
```

Comment:
Optional: it is unfortunate that the plural of `Series` is `Series`. I wonder if there's a better name for this function or if it's already obvious enough how `WaitForMetric` and `WaitForMetricSeries` differ based on the function signatures.